In [2]:
library(Stat2Data)
library(mosaic)
library(lattice)

data(MLB2007Standings)
mlb = MLB2007Standings
head(mlb)

Team                 League Wins Losses WinPct BattingAvg Runs Hits HR 
1 Arizona Diamondbacks NL     90   72     0.556  0.250      712  1350 171
2 Atlanta Braves       NL     84   78     0.519  0.275      810  1562 176
3 Baltimore Orioles    AL     69   93     0.426  0.272      756  1529 142
4 Boston Red Sox       AL     96   66     0.593  0.279      867  1561 166
5 Chicago Cubs         NL     85   77     0.525  0.271      752  1530 151
6 Chicago White Sox    AL     72   90     0.444  0.246      693  1341 190
  Doubles ⋯ RBI SB  OBP   SLG   ERA  HitsAllowed Walks StrikeOuts Saves WHIP
1 286     ⋯ 687 109 0.321 0.413 4.13 1446        546   1088       51    1.38
2 328     ⋯ 781  64 0.339 0.435 4.11 1442        537   1106       36    1.36
3 306     ⋯ 718 144 0.333 0.412 5.17 1491        696   1087       30    1.52
4 352     ⋯ 829  96 0.362 0.444 3.87 1350        482   1149       45    1.27
5 340     ⋯ 711  86 0.333 0.422 4.04 1340        573   1211       39    1.32
6 249     ⋯ 667  78 0.318 0.404 4.77 1556        499   1015       42    1.43

In [3]:
runsEraLm = lm(mlb$WinPct ~ mlb$Runs + mlb$ERA)
summary(runsEraLm)


Call:
lm(formula = mlb$WinPct ~ mlb$Runs + mlb$ERA)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.038586 -0.015104 -0.003181  0.013915  0.059104 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.103e-01  7.582e-02   6.730 3.16e-07 ***
mlb$Runs     5.113e-04  7.003e-05   7.301 7.47e-08 ***
mlb$ERA     -9.138e-02  1.181e-02  -7.741 2.52e-08 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.02604 on 27 degrees of freedom
Multiple R-squared:  0.8071,	Adjusted R-squared:  0.7928 
F-statistic: 56.48 on 2 and 27 DF,  p-value: 2.252e-10


### Regression Equation: $\hat{WinPct} = .5103 + 5.113\mathrm{e}{-4}*\text{Runs} - 9.138\mathrm{e}{-2}*\text{ERA}$

In [4]:
anova(runsEraLm)

Df Sum Sq     Mean Sq      F value  Pr(>F)      
mlb$Runs   1 0.03596578 0.0359657753 53.03620 7.821719e-08
mlb$ERA    1 0.04063441 0.0406344097 59.92071 2.521954e-08
Residuals 27 0.01830968 0.0006781364       NA           NA

In [6]:
0.01830968 / 27 # Mean sq = sse / dof -> 1 per parameter estimated, n-1 for the error terms

[1] 0.0006781363

In [8]:
# Computing a prediction interval for the Texas Rangers
# Need to figure out how this happened :(
TexasRangersData = data.frame(Runs=816, ERA=4.75)
predict(runsEraLm, TexasRangersData, interval="predict") 

Warning message:
“'newdata' had 1 row but variables found have 30 rows”

fit       lwr       upr      
1  0.4968959 0.4411913 0.5526004
2  0.5488284 0.4936448 0.6040120
3  0.4243542 0.3673158 0.4813926
4  0.5999029 0.5422588 0.6575469
5  0.5255713 0.4701810 0.5809615
6  0.4286969 0.3725450 0.4848488
7  0.4591766 0.4036391 0.5147141
8  0.5548226 0.4993828 0.6102625
9  0.5552018 0.4994962 0.6109075
10 0.5461606 0.4895507 0.6027706
11 0.4627555 0.4071949 0.5183162
12 0.4522595 0.3971312 0.5073879
13 0.4618444 0.4065672 0.5171216
14 0.5439978 0.4890149 0.5989808
15 0.5022584 0.4472354 0.5572815
16 0.5168122 0.4623763 0.5712482
17 0.4981359 0.4426380 0.5536338
18 0.5320534 0.4773824 0.5867244
19 0.5948845 0.5340529 0.6557161
20 0.4980155 0.4432711 0.5527598
21 0.5340958 0.4769759 0.5912157
22 0.4299252 0.3739114 0.4859391
23 0.5510173 0.4934127 0.6086219
24 0.4765860 0.4202185 0.5329535
25 0.4839909 0.4292399 0.5387419
26 0.4560236 0.4010003 0.5110469
27 0.4047497 0.3445926 0.4649067
28 0.4934113 0.4383737 0.5484489
29 0.5297378 0.4741697 0.5853060
30 0.4358341 0.3794122 0.4922559

In [11]:
data(faithful)
attach(faithful)
eruption.lm = lm(eruptions ~ waiting)
newdata = data.frame(waiting=80)
predict(eruption.lm, newdata, interval="predict")

fit     lwr      upr     
1 4.17622 3.196089 5.156351